### PR TITLE
Add jellyfish as its own tool

### PR DESCRIPTION
**Pull request by astrovsky01**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)
Though I haven't put in all functionality yet, I intend to. it will be used in a workshop, so the functions currently wrapped are the most important. I will continue to update, but wanted to the necessary ones in first.

### CODE REVIEW

File: tools/jellyfish/jellyfish.xml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+                <option value="histo"/>
+                <option value="dump"/>
+                <!-- <option value="bc"/>
+                <option value="info"/>
+                <option value="stats"/>
+                <option value="merge"/>
+                <option value="query"/> -->
+            </param>
+            <when value="count">
+                <param name="input" type="data" format="fasta,fastq,fastqsanger" label="Input data file"/>
```
Comment (bgruening) [inline tools/jellyfish/jellyfish.xml:101]: can jellyfish consume fasta.gz and fastq.gz files?

File: tools/jellyfish/jellyfish.xml [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                <param name="mer_len" argument="--mer-len" type="integer" label="Length of mer" value="20" min="1"/>
+                <param name="size" argument="--size" type="integer" label="initial hash size" value="32" min="1"/>
```
Comment (bgruening) [inline tools/jellyfish/jellyfish.xml:103]: when you have argument="--size" you can remove the `name`

File: tools/jellyfish/jellyfish.xml [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                    --bf-fp $commands.bloom.bf_fp
+                #end if
+            #else if $commands.bloom.filters == "bf_size":
+                --bf-size $commands.bloom.bf_size
+                #if $commands.bloom.bf_fp:
+                    --bf-fp $commands.bloom.bf_fp
+                #end if
+            #end if
+            #if $commands.if:
+                --if $commands.if
```
Comment (bgruening) [inline tools/jellyfish/jellyfish.xml:31]: single quotes :)

File: tools/jellyfish/jellyfish.xml [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+            #if $commands.text == "text":
+                --text
+            #end if
+            #if $commands.lower:
+                --lower-count $commands.lower
+            #end if
+            #if $commands.upper:
+                --upper-count $commands.upper
+            #end if
+            $commands.input
```
Comment (bgruening) [inline tools/jellyfish/jellyfish.xml:45]: singe quotes

File: tools/jellyfish/jellyfish.xml [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+                --low $commands.low
+            #end if
+            #if $commands.high:
+                --high $commands.high
+            #end if
+            #if $commands.increment:
+                --increment $commands.increment
+            #end if
+            $commands.full
+            $commands.input
```
Comment (bgruening) [inline tools/jellyfish/jellyfish.xml:58]: single-quotes

File: tools/jellyfish/jellyfish.xml (hunk 1 above)
Comment (astrovsky01) [reply tools/jellyfish/jellyfish.xml:101]: Just tried, it can't

File: tools/jellyfish/jellyfish.xml [hunk 6]
```diff
@@ -0,0 +1,255 @@
+<tool id="jellyfish" name="jellyfish" version="@WRAPPER_VERSION@+@VERSION_SUFFIX@" profile="20.01">
+    <macros>
+        <token name="@WRAPPER_VERSION@">2.2.10</token>
+        <token name="@VERSION_SUFFIX@">galaxy0</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="2.3.0">kmer-jellyfish</requirement>
```
Comment (bernt-matthias) [inline tools/jellyfish/jellyfish.xml:7]: use @WRAPPER_VERSION@ here.

**Review by bernt-matthias — COMMENTED** [review]
Seems that jellyfish generates its CLI using yaggo https://github.com/gmarcais/yaggo

Wondering if its possible to parse the yaggo files and autogenerate the Galaxy wrapper, e.g. 
https://github.com/gmarcais/Jellyfish/blob/master/sub_commands/merge_main_cmdline.yaggo

File: tools/jellyfish/jellyfish.xml (hunk 6 above)
Comment (astrovsky01) [reply tools/jellyfish/jellyfish.xml]: Good catch, thanks

**Comment by astrovsky01** [comment]
ping @bgruening ready for merge

File: tools/jellyfish/jellyfish.xml [hunk 7]
```diff
@@ -112,40 +112,40 @@
...
+                        <param argument="--bf-fp" type="float" label="False positive rate of bloom filter" optional="true"/>
                     </when>
                     <when value="bf_size">
                         <param name="bf_size" argument="--bf-size" type="integer" label="Use bloom filter to count high-frequency mers"  min="1" optional="true"/>
-                        <param name="bf_fp" argument="--bf-fp" type="float" label="False positive rate of bloom filter" optional="true"/>
+                        <param argument="--bf-fp" type="float" label="False positive rate of bloom filter" optional="true"/>
                     </when>
                 </conditional>
-                <param name="if" argument="--if" type="data" format="txt" label="Count only k-mers in this file" optional="true"/>
-                <param name="reprobes" argument="--repropres" type="integer" label="Maximum number of repropbes" min="1" optional="true"/>
```
Comment (bgruening) [inline tools/jellyfish/jellyfish.xml]: @astrovsky01 there have been some typos `reprobes` vs. `repropres`. Thats why name should be remove when you have argument.

Comment (astrovsky01) [reply tools/jellyfish/jellyfish.xml]: 👍

**Review by bgruening — APPROVED** [review]

**Pull request merged**